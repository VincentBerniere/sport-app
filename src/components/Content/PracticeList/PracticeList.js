import React, { Component } from 'react';
import './PracticeList.scss';
import PracticeItem from './PracticeItem/PracticeItem';

class PracticeList extends Component {
    renderPracticeItems() {
        if (!this.props.practices || this.props.practices.length <= 0)
            return '';
        
        return this.props.practices.map(practice => {
            return (
                <PracticeItem key={practice.id} practice={practice} />
            );
        });
    }

    render() {
        return (
            <div className="ui relaxed divided list">
                <h4 className="ui header">{!this.props.practices || this.props.practices.length <= 1 ? 'Excercice' : 'Excercices'} :</h4>

                {this.renderPracticeItems()}
            </div>
        );
    }
}

PracticeList.defaultProps = {
    practices: []
}

export default PracticeList;
