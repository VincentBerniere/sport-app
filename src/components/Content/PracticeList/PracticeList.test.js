import React from 'react';
import { shallow } from 'enzyme'; 
import PracticeList from './PracticeList';

describe('PracticeList', () => {
    const practiceList = shallow(<PracticeList />);

    it('renders correctly', () => {
        expect(practiceList).toMatchSnapshot();
    });
});