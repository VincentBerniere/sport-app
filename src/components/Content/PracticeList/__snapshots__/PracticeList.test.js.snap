// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PracticeList renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PracticeList
    practices={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="ui header"
        >
          Excercice
           :
        </h4>,
        "",
      ],
      "className": "ui relaxed divided list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Excercice",
            " :",
          ],
          "className": "ui header",
        },
        "ref": null,
        "rendered": Array [
          "Excercice",
          " :",
        ],
        "type": "h4",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="ui header"
          >
            Excercice
             :
          </h4>,
          "",
        ],
        "className": "ui relaxed divided list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Excercice",
              " :",
            ],
            "className": "ui header",
          },
          "ref": null,
          "rendered": Array [
            "Excercice",
            " :",
          ],
          "type": "h4",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
